File: lib/src/router/route_handler.dart
library angular2.src.router.route_handler;

import "package:angular2/src/facade/async.dart" show Future, PromiseWrapper;
import "package:angular2/src/facade/lang.dart" show Type;

abstract class RouteHandler {
  Type componentType;
  Future<dynamic> resolveComponentType();
  Map<String, dynamic> data;
}

File: ts/src/router/route_handler.ts
import {Promise, PromiseWrapper} from 'angular2/src/facade/async';
import {Type} from 'angular2/src/facade/lang';
import {RouteData} from './instruction';

export interface RouteHandler {
  componentType: Type;
  resolveComponentType(): Promise<any>;
  data: RouteData;
}
